server/src/main/java/org/elasticsearch/search/aggregations/AggregationExecutionContext.java

```diff
@@ -23,11 +23,17 @@
 public class AggregationExecutionContext {
 
     private final CheckedSupplier<BytesRef, IOException> tsidProvider;
+    private final CheckedSupplier<Long, IOException> timestampProvider;
     private final LeafReaderContext leafReaderContext;
 
-    public AggregationExecutionContext(LeafReaderContext leafReaderContext, CheckedSupplier<BytesRef, IOException> tsidProvider) {
+    public AggregationExecutionContext(
+        LeafReaderContext leafReaderContext,
+        CheckedSupplier<BytesRef, IOException> tsidProvider,
+        CheckedSupplier<Long, IOException> timestampProvider
```

Comment:
Why is this a checked supplier? Under what condition can it ever throw IOException?

Reply:
> LGTM, except I don't think we need IOException there. I suspect we can convert CheckedSupplier to Supplier in case of tsid as well. I think it is CheckedSupplier for purely historical reasons.

Yeah, I also think Supplier is better, I have converted CheckedSupplier to Supplier for timestamp and tsid.